Synonyms APIs - Synonyms Sets documentation (#98015)
elastic/elasticsearch: docs/reference/synonyms/apis/put-synonyms-set.asciidoc

[[put-synoyms-set]]
=== Put Synonyms Set

beta::[]

++++
<titleabbrev>Put Synonyms Set</titleabbrev>
++++

Creates or updates a Synonyms Set.

[[put-synonyms-set-request]]
==== {api-request-title}

`PUT _synonyms/<synonyms_set>`

[[put-synonyms-set-prereqs]]
==== {api-prereq-title}

Requires the `manage_search_synonyms` cluster privilege.

[[put-synonyms-set-path-params]]
==== {api-path-parms-title}

`<synonyms_set>`::
(Required, string)
Synonyms set identifier to create.

> This identifier will be used by other Synonyms Management APIs to manage the synonyms set.

Reviewer: Link to the top level APIs page here?

Author: Nice!

`<body>`::
(Required, object)

> Contains the definition of the synonym rules that conform the synonyms set:

Reviewer 2: ```suggestion
Contains the definition of the synonym rules for this synonyms set:
```

> Contains the definition of the synonym rules that conform the synonyms set:

Reviewer 3: What does it mean  "synonym rules that conform the synonyms set:"?

Author: Chris had a better suggestion: "Contains the definition of the synonym rules for this synonyms set"

----
PUT _synonyms/my-synonyms-set
{
  "synonyms_set": [
    {
      "id": "test-1",
      "synonyms": "hello, hi"
    },
    {
      "synonyms": "bye, goodbye"
    },
    {
      "id": "test-2",
      "synonyms": "test => check"
    }
  ]
}
----
// TEST[skip:TBD]

If any of the synonym rules included is not valid, the API will return an error.

[source,console]
----
PUT _synonyms/my-synonyms-set
{
  "synonyms_set": [
    {
      "synonyms": "hello => hi => howdy"
    }
  ]
}
----
// TEST[skip:TBD]

[source,console-result]
----
{
  "error": {
    "root_cause": [
      {
        "type": "action_request_validation_exception",
        "reason": "Validation Failed: 1: More than one explicit mapping specified in the same synonyms rule: [hello => hi => howdy];"
      }
    ],
    "type": "action_request_validation_exception",
    "reason": "Validation Failed: 1: More than one explicit mapping specified in the same synonyms rule: [hello => hi => howdy];"
  },
  "status": 400
}
----
// TEST[skip:TBD]

[discrete]
==== Analyzer reloading
When an existing synonyms set is updated, the <<search-analyzer, search analyzers>> that use the synonyms set are reloaded automatically for all indices.

> This would be equivalent to invoke <<indices-reload-analyzers>> for all indices that use the synonyms set.

Reviewer 2: ```suggestion
This would be equivalent to invoking <<indices-reload-analyzers>> for all indices that use the synonyms set.
```

[[put-synonyms-set]]

> === Put synonyms set

Reviewer 4: ```suggestion
=== Create or update synonyms set
```

Reviewer 4: The convention has been to say "create or update" instead of "put" (see https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-put-template.html for example)

beta::[]

++++
<titleabbrev>Put synonyms set</titleabbrev>
++++

Creates or updates a synonyms set.

[[put-synonyms-set-request]]
==== {api-request-title}

`PUT _synonyms/<synonyms_set>`

[[put-synonyms-set-prereqs]]
==== {api-prereq-title}

Requires the `manage_search_synonyms` cluster privilege.

[[put-synonyms-set-path-params]]

> ==== {api-path-parms-title}

Reviewer 4: I think it would be better to have a section for query parameters and a section for the request body here. See https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-put-template.html#put-index-template-api-path-params for example.

Author: Thanks for the link, changing it!

`<synonyms_set>`::
(Required, string)
Synonyms set identifier to create.
This identifier will be used by other <<synonyms-apis>> to manage the synonyms set.

`<body>`::
(Required, object)
Contains the definition of the synonym rules for this synonyms set:
+
.Properties of `<body>` objects
[%collapsible%open]
====
`synonyms_set`::
(Required, array of synonym rules objects)
The synonym rules definitions for the synonyms set.

.Properties of `synonyms_set` objects
[%collapsible%open]
=====

`id`::
(Optional, string)
// TODO link to synonym rules APIs
The identifier associated to the synonym rule, that can be used to manage individual synonym rules via Synonym Rules APIs.
In case a synonym rule id is not specified, an identifier will be created automatically by the API.

> 

Reviewer 3: by the API =>  by {es} 

We usually say "by Elasticsearch"
